src/backend/Csrs.Api/Features/UserRequests/Create.cs

```diff
@@ -1,27 +1,126 @@
+
+                string userId = _userService.GetBCeIDUserId();
+                //string userId = "44e88bfd-fed0-4a68-97f7-0f701df51315";
+                if (string.IsNullOrEmpty(userId))
+                {
+                    _logger.LogInformation("No BCeID on authenticated user, cannot create account");
+                    return new Response("Unauthenticated");
+                }
+                var bceidScope = _logger.AddBCeIdGuid(userId);
+                // find to see if the person has an account already?
+                MicrosoftDynamicsCRMssgCsrspartyCollection parties = await _dynamicsClient.GetPartyByBCeIdAsync(userId, cancellationToken);
+                MicrosoftDynamicsCRMssgCsrsparty party;
+                if (parties.Value.Count > 0)
+                {
+                    party = parties.Value.First();
+                }
+                else
+                {
+                    _logger.LogInformation("No associated party, cannot create User Request");
+                    return new Response("No Party Associated");
+                }
+
+                _logger.LogInformation("Creating User Request");
+                MicrosoftDynamicsCRMactivitypointer ap = new MicrosoftDynamicsCRMactivitypointer();
+                //activitytypecode to task.
+                ap.Activitytypecode = "task";
+                ap.Subject = "File " + request.FileNo + " - " + request.RequestType;
+                string desc = "Party: " + party.SsgBceidDisplayname + "/n"+
+                              "Message: "+request.RequestMessage;
+                ap.Description =  desc;
+                ap.Isregularactivity = true;
+                //Get the file
+                MicrosoftDynamicsCRMssgCsrsfile originFile;
+                try
+                {
+                    originFile = await _dynamicsClient.GetFileByFileNumber(request.FileNo, cancellationToken);
+                }
+                catch (HttpOperationException exception) when (exception.Response?.StatusCode == HttpStatusCode.NotFound)
+                {
+                    _logger.LogInformation("No file associated, cannot create User Request");
+                    return new Response("Incorrect file number supplied");
+                }
+                if (originFile != null)
+                {
+                    //This OWNERID relationship IS DIFFERENT AND DOES USE select systemusers by ownerid
+                    //SystemUser is just a child of Principal
+                    //This was verified by dynamics team and does not work correct if you principals
+                    //As per Burak/Dynmiacs team we should never touch principals. 
+                    ap.OwnerIdODataBind = _dynamicsClient.GetEntityURI("systemusers", originFile._owneridValue);
+                }
+                ap.RegardingobjectidSsgCsrsfileODataBind = _dynamicsClient.GetEntityURI("systemusers", originFile.SsgCsrsfileid);
```

Comment:
this looks odd, using systemusers entity but passing a file id

Reply:
The ownerId is not the fileid that is ssg_csrsfileid.  ownerId is the Principal which according to Burak.    I noted it was strange too.  That was a fun one to debug.   I also tried using OwningUser instead but that dont work as in Dynamics a file can belong to a user or group. When you assign both options there.